enterprise/reporting/error_extractor.go

```diff
@@ -343,13 +343,50 @@ func getErrorCodeFromStatTags(statTags map[string]string) string {
 func (ext *ExtractorHandle) isVersionDeprecationError(errorMessage string) bool {
 	var score int
 
+	// Convert to lowercase for case-insensitive matching
 	errorMessage = strings.ToLower(errorMessage)
+
+	// 1. Check for direct keyword matches (existing approach)
 	for keyword, s := range lowercasedDeprecationKeywords {
 		if strings.Contains(errorMessage, keyword) {
 			score += s
 		}
 	}
 
+	// 2. Check for version-related patterns with negative context
+	if score <= ext.versionDeprecationThresholdScore.Load() {
+		// Version indicators
+		versionPatterns := []string{"version", "api", "endpoint"}
+
+		// status indicators
+		statusPatterns := []string{
+			"not.*active", "inactive", "no longer.*active",
+			"not.*supported", "unsupported", "no longer.*supported",
+			"not.*valid", "invalid", "no longer.*valid",
+			"not.*available", "unavailable", "no longer.*available",
+			"disabled", "expired", "removed", "discontinued",
+			"deprecated", "obsolete",
+		}
```

Comment:
Since these seem static, should we make it global instead of creating it every time?

Reply:
done